src/pages/workspace/companyCards/WorkspaceCompanyCardsPage.tsx

```diff
@@ -108,47 +100,68 @@ type WorkspaceCompanyCardPageProps = StackScreenProps<FullScreenNavigatorParamLi
 
 function WorkspaceCompanyCardPage({route}: WorkspaceCompanyCardPageProps) {
     const {translate} = useLocalize();
+    const styles = useThemeStyles();
+    const theme = useTheme();
     const policyID = route.params.policyID;
-    // const workspaceAccountID = PolicyUtils.getWorkspaceAccountID(policyID);
-
-    // TODO: use data form onyx instead of mocked one when API is implemented
-    // const [cardFeeds] = useOnyx(`${ONYXKEYS.COLLECTION.SHARED_NVP_PRIVATE_DOMAIN_MEMBER}${workspaceAccountID}`);
-    const cardFeeds = mockedFeeds;
+    const policy = usePolicy(policyID);
+    const workspaceAccountID = policy?.workspaceAccountID ?? -1;
+    const [cardFeeds] = useOnyx(`${ONYXKEYS.COLLECTION.SHARED_NVP_PRIVATE_DOMAIN_MEMBER}${workspaceAccountID.toString()}`);
     const [lastSelectedFeed] = useOnyx(`${ONYXKEYS.COLLECTION.LAST_SELECTED_FEED}${policyID}`);
     const defaultFeed = Object.keys(cardFeeds?.companyCards ?? {})[0];
     const selectedFeed = lastSelectedFeed ?? defaultFeed;
+    const fetchCompanyCards = useCallback(() => {
+        Policy.openPolicyCompanyCardsPage(policyID, workspaceAccountID);
+    }, [policyID, workspaceAccountID]);
+
+    useFocusEffect(fetchCompanyCards);
+
+    const companyCards = cardFeeds?.companyCards ?? {};
+    const selectedCompanyCard = companyCards[selectedFeed] ?? {};
```

Comment:
If you do `?? null` then you don't need the isEmpty check below, you can just do `!selectedCompanyCard`

Reply:
makes sense